docs/src/snap/howto/epa.md

```diff
@@ -0,0 +1,1061 @@
+
+# enable kernel options
+- update-grub
+
+# reboot to activate realtime-kernel and kernel options
+power_state:
+  mode: reboot
+```
+
+Notes:
+
+* In the above, realtime kernel 6.8 is installed from a private ppa. It was recently backported from 24.04 to 22.04 and is still going through some validation stages. Once it is officially released, it will be installable via the Ubuntu Pro cli.
+
+
+````
+`````
+
+
+
+
+
+## {{product}} setup 
+
+{{product}} is delivered as a 
+[snap](https://documentation.ubuntu.com/canonical-kubernetes/latest/snap/).
+
+This section explains how to set up a dual node {{product}} cluster for testing
+EPA capabilities.
+
+### Control plane and worker node 
+
+1. [Install the
+   snap](https://documentation.ubuntu.com/canonical-kubernetes/latest/snap/howto/install/snap/)
+   from the moonray track. The [beta
+   channel](https://documentation.ubuntu.com/canonical-kubernetes/latest/snap/explanation/channels/)
+   is used at this point as the end configuration of the k8s snap is not
+   finalised yet.
+
+```
+sudo snap install k8s --classic --channel=1.30-moonray/beta
+```
+
+2. Create a file called *configuration.yaml*. In this configuration file we let
+   the snap start with its default CNI (calico), with CoreDNS deployed and we
+   also point k8s to the external etcd. 
+
+```
+cluster-config:
+  network:
+    enabled: true
+  dns:
+    enabled: true  local-storage:
+    enabled: true
+extra-node-kubelet-args:
+  --reserved-cpus: "0-31"
+  --cpu-manager-policy: "static"
+  --topology-manager-policy: "best-effort"
+```
+
+3. Bootstrap {{product}} using the above configuration file.
+
+```
+sudo k8s bootstrap --file configuration.yaml
+```
+
+#### Verify control plane node is up
+
+After a few seconds you can query the API server with:
+
+```
+sudo k8s kubectl get all -A
+```
+
+### Second k8s node as worker 
+
+1. Install the k8s snap on the second node
+
+```
+sudo snap install k8s --classic --channel=1.30-moonray/beta
+```
+
+2. On the control plane node generate a join token to be used for joining the
+   second node
+
+```
+sudo k8s get-join-token --worker
+```
+
+3. On the worker node create the configuration.yaml file
+
+```
+extra-node-kubelet-args:
+  --reserved-cpus: "0-31"
+  --cpu-manager-policy: "static"
+  --topology-manager-policy: "best-effort"
+```
+
+4. On the worker node use the token to join the cluster
+
+```
+sudo k8s join-cluster --file configuration.yaml <token-generated-on-the-control-plane-node>
+```
+
+
+### Verify the two node cluster is ready 
+
+After a few seconds the second worker node will register with the control
+plane. You can query the available workers from the first node:
+
+```
+sudo k8s kubectl get nodes
+```
+
+The output should list the connected nodes:
+
+```
+NAME          STATUS   ROLES                  AGE   VERSION
+pc6b-rb4-n1   Ready    control-plane,worker   22h   v1.30.2
```

Comment:
Should we use k8s v1.31 for the docs?

Reply:
I will make both {{verion}} and {{track}} variables we can use in the text, but it is hard to change the ouput that way. i think we can just manually edit to whatever we want. I guess I will also need to find a way of tagging these things so we know to update them